src/js/fable-core/Array2.fs

```diff
@@ -8,9 +8,25 @@ type ArrayCons =
     [<Emit("$0.from($1)")>]
     abstract From: 'T seq -> 'T array
 
-let map (f: 'T->'U) (source: 'T[]) (cons: ArrayCons) =
+let mapIndexed (f: int -> 'T -> 'U) (source: 'T[]) (cons: ArrayCons) =
+    let len = source.Length
+    let target = cons.Create(len)
+    for i = 0 to (len - 1) do
+        target.[i] <- f i source.[i]
+    target
+
+let map (f: 'T -> 'U) (source: 'T[]) (cons: ArrayCons) =
+    // TODO: Should we involve currying and re-use mapIndexed here?
+    // Don't quite know yet how is it optimized so can't decide on performance vs code reuse.
+    // mapIndexed (fun _ x -> f x) source
+
     let len = source.Length
     let target = cons.Create(len)
     for i = 0 to (len - 1) do
         target.[i] <- f source.[i]
     target
+
+let indexed (source: 'T[]) (cons: ArrayCons) =
+    // TODO: How do we get concrete instance of ArrayCons for dynamic JS Arrays creation?
+    // Passing as a parameter for now.
```

Comment:
Fable knows this will return `int*'T[]` so `cons` should already be a dynamic JS Array constructor.

Reply:
OK, thx, got it. What about functions like `partition` returning `'T[] * 'T[]`? It's in the list of "functions not returning an array", by the way. I believe it should be handled by do passing `cons`, or am I still missing something here?

Reply:
It was in the list of functions already implemented in Array.ts, but you're right, it needs the constructor to be passed. I moved it to the first group, thanks!